pkg/agent/kubeagent.go

```diff
@@ -126,3 +169,261 @@ func (k *KubeAgent) GetFactories(_ context.Context) (otelcol.Factories, error) {
+	case DaemonSet:
+		daemonSet, err := c.Clientset.AppsV1().DaemonSets(c.AgentNamespace).Get(context.TODO(), c.Daemonset, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+
+		daemonSet.Spec.Template.ObjectMeta.Labels[Timestamp] = fmt.Sprintf("%d", metav1.Now().Unix())
+		_, err = c.Clientset.AppsV1().DaemonSets(c.AgentNamespace).Update(context.TODO(), daemonSet, metav1.UpdateOptions{})
+		if err != nil {
+			return err
+		}
+
+	case Deployment:
+		deployment, err := c.Clientset.AppsV1().Deployments(c.AgentNamespace).Get(context.TODO(), c.Deployment, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+
+		deployment.Spec.Template.ObjectMeta.Labels[Timestamp] = fmt.Sprintf("%d", metav1.Now().Unix())
+		_, err = c.Clientset.AppsV1().Deployments(c.AgentNamespace).Update(context.TODO(), deployment, metav1.UpdateOptions{})
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// updateConfigMap gets the latest configmap from Middleware backend and updates the k8s configmap
+// based on component type
+func (c *KubeAgentMonitor) updateConfigMap(componentType ComponentType) error {
+
+	u, err := url.Parse(c.APIURLForConfigCheck)
+	if err != nil {
+		return err
+	}
+
+	baseURL := u.JoinPath(apiPathForYAML)
+	baseURL = baseURL.JoinPath(c.APIKey)
+	params := url.Values{}
+	params.Add("platform", "k8s")
+	params.Add("component_type", componentType.String())
+	params.Add("cluster", os.Getenv("MW_KUBE_CLUSTER_NAME"))
+
+	// Add Query Parameters to the URL
+	baseURL.RawQuery = params.Encode() // Escape Query Parameters
+
+	resp, err := http.Get(baseURL.String())
+	if err != nil {
+		c.logger.Error("failed to call Restart-API", zap.String("url", baseURL.String()), zap.Error(err))
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		c.logger.Error("failed to call get configuration api", zap.Int("statuscode", resp.StatusCode))
+		return ErrRestartStatusAPINotOK
+	}
+
+	// Read response body
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		c.logger.Error("failed to reas response body", zap.Error(err))
+		return err
+	}
+
+	// Unmarshal JSON response into ApiResponse struct
+	var apiResponse apiResponseForYAML
+	if err := json.Unmarshal(body, &apiResponse); err != nil {
+		c.logger.Error("failed to unmarshal api response", zap.Error(err))
+		return err
+	}
+
+	// Verify API Response
+	if !apiResponse.Status {
+		c.logger.Error("failure status from api response for ingestion rules", zap.Bool("status", apiResponse.Status))
+		return ErrInvalidResponse
+	}
+
+	var apiYAMLConfig map[string]interface{}
+	if len(apiResponse.Config.DaemonSet) == 0 && len(apiResponse.Config.Deployment) == 0 {
+		c.logger.Error("failed to get valid response",
+			zap.Int("config docker len", len(apiResponse.Config.Docker)),
+			zap.Int("config no docker len", len(apiResponse.Config.NoDocker)))
+		return ErrInvalidResponse
+	}
+
+	// apiYAMLConfig = apiResponse.Config.NoDocker
+	if componentType == DaemonSet {
```

Comment:
Golang way to write this would be

```
apiYAMLConfig = apiResponse.Config.Deployment
if componentType == DaemonSet {
	apiYAMLConfig = apiResponse.Config.DaemonSet
} 
```